apps/publisher/extensions/assets/default/apis/assets.jag

```diff
@@ -43,170 +93,143 @@ require('/modules/publisher.js').exec(function (ctx) {
     var ASSET_STATE_URL = '/{context}/apis/assets/{id}/state';
-
     var method = req.getMethod();
     var options = ref.getQueryOptions(req.getQueryString());
-    var response = ctx.response;
-    response.contentType = 'application/json';
-
-    var constants = require("rxt").constants;
-
+    var response;
+    //response.contentType = 'application/json';
+    var constants = require('rxt').constants;
+    var contentTypeheader, jsonHeader, multipartHeader, formHeader, acceptHeader, jsonHeaderAccept;
     switch (method) {
         case 'POST':// POST endpoints
             var newAsset;
             if (uriMatcher.match(CREATE_URL)) {//to create asset
+                contentTypeheader = req.getHeader('Content-Type');
+                jsonHeader = contentTypeheader.indexOf('application/json');
+                multipartHeader = contentTypeheader.indexOf('multipart/form-data');
+                formHeader = contentTypeheader.indexOf('application/x-www-form-urlencoded');
                 try {
-                    newAsset = assetApi.create(options, req, res, session);
-                    //If the asset is successfully created then return the asset
-                    if (newAsset) {
-                        response = responseProcessor.buildSuccessResponse(res,
-                                constants.STATUS_CODES.CREATED, newAsset);
-                        // We set this as a helper to alert the user if the asset creation time is long
-                        //session.put("latest-asset",options.attributes.overview_name);
-                    } else {
-                        response = responseProcessor.buildErrorResponse(res,
-                                constants.STATUS_CODES.INTERNAL_SERVER_ERROR, 'Failed to create asset of type: ' +
-                                        options.type);
+                    if (!contentTypeheader) {
+                        response = responseProcessor.buildErrorResponseDefault(constants.STATUS_CODES.UNSUPPORTED_MEDIATYPE, 'error on create asset', response, 'Unsupported Media Type of creating ' + options.type, 'more information on create asset error', []);
+                    }
+                    if (jsonHeader >= 0 || multipartHeader >= 0 || formHeader >= 0) {
+                        addAsset(options, req, res, session);
                     }
+                    response = responseProcessor.buildErrorResponseDefault(constants.STATUS_CODES.UNSUPPORTED_MEDIATYPE, 'error on create asset', response, 'Unsupported Media Type of creating ' + options.type, 'more information on create asset error', []);
                 } catch (e) {
-                    log.error(e);
-                    response = responseProcessor.buildErrorResponse(res,
-                            constants.STATUS_CODES.INTERNAL_SERVER_ERROR, 'Failed to create asset of type: ' +
-                                    options.type);
+                    log.error(e.message, e);
```

Comment:
Additional thing, when an error is logged, if you have any additional context aware information, construct a meaningful message for the person who reads the log. Then he can easily identify the error. For any additional information, you can use debug logs with isDebugEnabled() check. E.g. asset content etc. can also be logged as debug. But don't log it a error, since it has too many data. For the error, either the name or an ID would be enough.